### PR TITLE
Account for pagination of datasources, calculations & collections (server-side)

### PR DESCRIPTION
**Pull request by Valexr**

### CODE REVIEW

File: routes/[version]/datasources/[id]/filters.js [hunk 1]
```diff
@@ -2,14 +2,18 @@ const { StatusCodes } = require('http-status-codes');
...
+const { getAndPrepareCollections } = require('../../collections/_helpers');
 
 module.exports = {
     patch: [
         checkAuth,
         patch,
         getUserCollections,
-        (req, res) =>
-            res.sse.sendTo({ reqId: req.id, data: req.session.collections }, 'collections'),
+        async (req, res) => {
```
Comment (blokhin) [inline routes/[version]/datasources/[id]/filters.js:12]: @Valexr `async` in the exports, I don't see it often, sure?

**Review by blokhin — COMMENTED** [review]
@Valexr please remove `[routes/[version]/auth/custom_webhook.js` it's a relic of my home-baked API implementation

**Comment by Valexr** [comment]
> @Valexr please remove `[routes/[version]/auth/custom_webhook.js` it's a relic of my home-baked API implementation

Ok

File: config.js [hunk 2]
```diff
@@ -34,6 +36,19 @@ module.exports = {
...
+            clientID: 'basf',
+            clientSecret: 'basf',
+            callbackURL: 'http://localhost:3000/v0/auth/basf',
+        } : {
+            authorizationURL: process.env.BASF_AUTH_URL || oauth.basf.auth_url || 'unset', // ???
+            tokenURL: process.env.BASF_TOKEN_URL || oauth.basf.token_url || 'unset', // ???
+            clientID: process.env.BASF_CLIENT_ID || oauth.basf.client || 'unset',
+            clientSecret: process.env.BASF_CLIENT_SECRET || oauth.basf.secret || 'unset',
+            callbackURL: process.env.BASF_CALLBACK_URL || oauth.basf.callback || 'unset',
+        }),
```
[the remark this answers is not in the document]
Comment (Valexr) [reply config.js:51]: Ok. I’m removing this

File: routes/[version]/calculations/[id].js [hunk 3]
```diff
@@ -21,13 +23,20 @@ async function del(req, res, next) {
     try {
         await deleteAndClearCalculation(req.user.id, req.params.id);
 
-        req.session.calculations = req.session.calculations.filter(
+        req.session.calculations.data = req.session.calculations.data.filter(
             (calculation) => calculation.id !== req.params.id
         );
 
+        console.log('req.session.calculations', req.session.calculations);
```
[the remark this answers is not in the document]
Comment (Valexr) [reply routes/[version]/calculations/[id].js:30]: No

File: routes/[version]/datasources/[id]/filters.js (hunk 1 above)
Comment (Valexr) [reply routes/[version]/datasources/[id]/filters.js:12]: Ok. I’m move this to separated function.

**Pull request merged**